packages/samples/headless-ssr/app/components/search-page.tsx

```diff
@@ -0,0 +1,24 @@
+'use client';
+
+import {SearchHydrationResult, hydrateInitialState} from '@/app/common/engine';
+import {useEffect, useState} from 'react';
+import ResultList from './result-list';
+
+// TODO: replace "any" with specific type
+export default function SearchPage({engineSnapshot}: {engineSnapshot: any}) {
+  const [hydrationResult, setHydrationResult] =
+    useState<SearchHydrationResult | null>(null);
+
+  useEffect(() => {
+    hydrateInitialState(engineSnapshot).then(({engine, controllers}) => {
+      setHydrationResult({engine, controllers});
+    });
+  }, [engineSnapshot]);
+
+  if (hydrationResult === null) {
+    return <div>Hydrating engine ..</div>;
```

Comment:
I think we should avoid having a loading state while the engine is hydrating, since this would cause the page to temporarily disappear (AFAIK).

Instead, the approach I would recommend is to temporarily render the engine/controller state from the `engineSnapshot`.

Reply:
Changed to use suspense when loading as it would be good to demonstrate using suspense and it keeps it simpler